apps/frontend/src/app/2_molecules/AmountRenderer/AmountRenderer.tsx

```diff
@@ -60,24 +61,29 @@ export const AmountRenderer: FC<AmountRendererProps> = ({
     });
   }, [addNotification, value]);
 
-  const formattedValue = useMemo(() => {
-    const numberValue = Number(value);
-    const decimals = getDecimalPartLength(numberValue);
+  const countUpValues = useMemo(() => {
+    let endValue = String(value);
 
-    if (decimals > precision) {
-      return Number(numberValue.toString().slice(0, -(decimals - precision))); // trimming the difference between the number of decimals we have and the precision we want, cannot use toFixed() as we need to round down
+    if (typeof value === 'number' && isScientificNumber(value)) {
+      endValue = value.toFixed(18);
     }
-    return numberValue;
+
+    const [whole = '', decimals = ''] = endValue.split('.');
```

Comment:
```suggestion
    const [whole = '', decimals = ''] = endValue.split(decimal);
```
You cannot hardcode a decimal separator like that, it will fail if you pass a localized string with some other decimal separator (in this case, try to pass `10,12` and you'll see). Use `decimal` from `getLocaleSeparators` instead or parse with a regex that accepts both `.` and `,` (but you may have problems with thousands separator).

Reply:
`endValue` is not localised but stringified number (for example using fromWei), so decimal always `.`

Reply:
But if you try to pass e.g. `10,12` to debt in LoC Status, it won't be parsed correctly.

Reply:
Checked all usages and I didn't see that there is localised strings passed to the component. And it shouldn't to my opinion. 
To make sure that dev does not pass it by mistake, maybe I could enforce dev to always use `BigNumber` instead with this task: https://sovryn.atlassian.net/browse/SOV-1988

Reply:
@creed-victor good idea, if there's not a million references to update then I say do it (unless tiltom objects)

Reply:
@creed-victor that sounds good